examples/plot_knockoff_aggregation.py

```diff
@@ -1,234 +1,216 @@
+)
+ax.legend(framealpha=0.2)
+# Plot the power
+ax = axes[1]
+sns.swarmplot(
+    data=df_plot,
+    y="Power",
+    ax=ax,
+)
+sns.despine()
+_ = plt.tight_layout()
 
-# %%
-# Aggregation methods provide a more stable inference
-# ---------------------------------------------------
-effect_number_samples(n_samples=n_samples)
 
 # %%
-# By repeating the model-X Knockoffs, we can see that instability
-# of the inference. Additionally, we can see that the aggregation method
-# is more stable. However, the e-values aggregation is more conservative,
-# i.e. the expect variables of importance is not find.
+# Aggregation procedures
+# ----------------------
+# We now compute the aggregation using both the p-values aggregation procedure from
+# :footcite:t:`pmlr-v119-nguyen20a` and the e-values aggregation procedure from
+# :footcite:t:`Ren_2023`. We then compare the results of both procedures in terms of
+# FDP and power.
+
+pval_aggregation = model_x_knockoff.fdr_selection(fdr=fdr, adaptive_aggregation=True)
+fdp_pval_agg, power_pval_agg = fdp_power(np.where(pval_aggregation)[0], non_zero_index)
+
+eval_aggregation = model_x_knockoff.fdr_selection(
+    fdr=fdr, fdr_control="ebh", evalues=True
+)
+fdp_eval_agg, power_eval_agg = fdp_power(np.where(eval_aggregation)[0], non_zero_index)
+
+df_plot["Method"] = "Individual KO"
+df_plot_2 = pd.concat(
+    [
+        df_plot,
+        pd.DataFrame(
+            {
+                "FDP": [fdp_pval_agg],
+                "Power": [power_pval_agg],
+                "Method": ["P-value aggregation"],
+            }
+        ),
+        pd.DataFrame(
+            {
+                "FDP": [fdp_eval_agg],
+                "Power": [power_eval_agg],
+                "Method": ["E-value aggregation"],
+            }
+        ),
+    ],
+    ignore_index=True,
+)
+
+# Plot the results
+# ----------------
+# In addition to the individual selections (blue), we plot the FDR and power obtained by
+# p-value aggregation (orange) and e-value aggregation (green).
+
+# sphinx_gallery_thumbnail_number = 2
+_, axes = plt.subplots(1, 2, figsize=(6, 3.5))
+ax = axes[0]
+sns.stripplot(
+    data=df_plot_2,
+    y="FDP",
+    hue="Method",
+    ax=ax,
+    palette="muted",
+    dodge=1,
+    legend=False,
+    size=8,
+    linewidth=1,
+)
+ax.axhline(fdr, color="tab:red", linestyle="--", lw=2, label="Desired FDR")
+
+ax = axes[1]
+sns.stripplot(
+    data=df_plot_2,
+    y="Power",
+    hue="Method",
+    ax=ax,
+    palette="muted",
+    dodge=True,
+    size=8,
+    linewidth=1,
+)
+sns.despine()
+_ = plt.tight_layout()
 
-# %%
-# Limitation of the aggregation methods
-# -------------------------------------
-effect_number_samples(n_samples=50)
 
 # %%
-# One important point of this method is that they require enough samples to
-# estimate the distribution of each feature.
+# It appears that both aggregation procedures successfully lowers the false discovery
+# proportion while maintaining a good statistical power.
```

Comment:
Actually, looking at the result, the p-values aggregation is powerless... is it just bad luck ?